extensions/wikia/WikiFactory/WikiFactory.php

```diff
@@ -1223,23 +1240,23 @@ static public function getLocalEnvURL( $url ) {
 		}
 
 		$server = str_replace( '.' . $devbox . '.wikia-dev.com', '', $server );
-		$server = str_replace( '.wikia.com', '', $server );
+		$server = str_replace( static::WIKIA_TOP_DOMAIN, '', $server );
 
 		// put the address back into shape and return
 		switch($wgWikiaEnvironment) {
 			case WIKIA_ENV_PREVIEW:
-				return 'http://preview.' . $server . '.wikia.com'.$address;
+				return 'http://preview.' . $server . static::WIKIA_TOP_DOMAIN . $address;
 			case WIKIA_ENV_VERIFY:
-				return 'http://verify.' . $server . '.wikia.com'.$address;
+				return 'http://verify.' . $server . static::WIKIA_TOP_DOMAIN . $address;
 			case WIKIA_ENV_STABLE:
-				return 'http://stable.' . $server . '.wikia.com'.$address;
+				return 'http://stable.' . $server . static::WIKIA_TOP_DOMAIN . $address;
 			case WIKIA_ENV_STAGING:
 			case WIKIA_ENV_PROD:
 				return sprintf( 'http://%s.%s%s', $server, $wgWikiaBaseDomain, $address ) ;
 			case WIKIA_ENV_SANDBOX:
-				return 'http://' . self::getExternalHostName() . '.' . $server . '.wikia.com' . $address;
+				return 'http://' . static::getExternalHostName() . '.' . $server . static::WIKIA_TOP_DOMAIN . $address;
 			case WIKIA_ENV_DEV:
-				return 'http://' . $server . '.' . self::getExternalHostName() . '.wikia-dev.com'.$address;
+				return 'http://' . $server . '.' . static::getExternalHostName() . '.wikia-dev.com'.$address;
 		}
 
 		throw new Exception( sprintf( '%s: %s', __METHOD__, 'unknown env detected' ) );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS112)